spdy_test.go

```diff
@@ -252,6 +252,91 @@ func TestHalfClose(t *testing.T) {
 	wg.Wait()
 }
 
+func TestUnexpectedRemoteConnectionClosed(t *testing.T) {
+	tt := []struct {
+		closeReceiver bool
+		closeSender   bool
+	}{
+		{closeReceiver: true, closeSender: false},
+		{closeReceiver: false, closeSender: true},
+		{closeReceiver: false, closeSender: false},
+	}
+	for tix, tc := range tt {
+		listen := "localhost:7743"
```

Comment:
Just pushed a change (forced update) to close the listener. Can you see if that helps?